server/src/main/java/org/hl7/davinci/endpoint/rems/database/requirement/MetRequirement.java

```diff
@@ -0,0 +1,107 @@
+package org.hl7.davinci.endpoint.rems.database.requirement;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.hl7.davinci.endpoint.rems.database.drugs.Drug;
+import org.hl7.davinci.endpoint.rems.database.fhir.RemsFhir;
+import org.hl7.davinci.endpoint.rems.database.rems.Rems;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.*;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+@Entity
+@Table(name = "metRequirement")
+public class MetRequirement {
+    @Id
+    @GeneratedValue
+    @Column(name = "id", nullable = false)
+    private Integer id;
+
+    @Column(name = "createdAt", nullable = false)
+    private String createdAt;
+
+    @Column(name = "completed", nullable = false)
+    private boolean completed ;
+
+    // FHIR resource which defines the requirement (task, questionnaire, etc)
+    @JoinColumn(name = "completedRequirement", nullable = true)
+    @OneToOne(fetch = FetchType.LAZY)
+    private RemsFhir completedRequirement;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name="REQUIREMENT_ID")
+    @JsonIgnore
+    private Requirement requirement;
+
+    @ManyToOne
+    @JoinColumn(name="REMS_REQUEST")
+    @JsonIgnore
+    private Rems remsRequest;
```

Comment:
This is missing a getter/setter, it needs to be set prior to the remsRequest being saved to the repository or the link between this metRequirement and the remsRequest won't be made, hence when reloading the response the UI shows that the metRequirements are empty.  It's because the metRequirements do not get linked and when the database tries to load the remsRequest it cannot find the associated metRequirements.  

The fix is to add a getter/setter for this here, and in RemsController, adding the remsRequest to the metRequirements before adding the metRequirement to the remsRequest.

Reply:
I'm still having some issues with loading in relationships with a select query to the db, but I don't think this is worth holding up this PR over since the release is scheduled for today. We have follow up work to integrate this db work into the rest of the workflow and will address the relationship loading at that point